tests/introspection/ApiSignatureTest.cs

```diff
@@ -865,5 +865,112 @@ protected virtual void CheckManagedMemberSignatures (MethodBase m, Type t, ref i
 					ReportError ($"`{t.Name}.{m.Name}` return type `{rt.Name}` is a concrete type `[Model]` and not an interface `[Protocol]`");
 			}
 		}
+
+		static bool IsDiscouraged (MemberInfo mi)
+		{
+			foreach (var ca in mi.GetCustomAttributes ()) {
+				switch (ca.GetType ().Name) {
+				case "ObsoleteAttribute":
+				case "AdviceAttribute":
+				case "ObsoletedAttribute":
+				case "DeprecatedAttribute":
+					return true;
+				}
+			}
+			return false;
+		}
+
+#if !MONOMAC
+		[Test]
+#endif
+		public void AsyncCandidates ()
+		{
+			int n = 0;
+			Errors = 0;
+			ErrorData.Clear ();
+
+			foreach (Type t in Assembly.GetTypes ()) {
+
+				// e.g. delegates used for events
+				if (t.IsNested)
+					continue;
+
+				if (!NSObjectType.IsAssignableFrom (t))
+					continue;
+
+				if (t.GetCustomAttribute<ProtocolAttribute> () != null)
```

Comment:
Weird that you got the ones from `NSFilePresenter` if you are actually skipping them 🤔